src/board.rs

```diff
@@ -487,7 +510,7 @@ impl Chessboard {
     /// value. Light squares are represented by (r: 190, g: 140, b: 170), and dark squares are
     /// represented by (r: 255, g: 206, b: 158).
     #[rustfmt::skip]
-    fn find_bkgnd(&self, rank: usize, file: usize) -> Color {
+    fn find_bkgnd(&self, rank: u8, file: u8) -> Color {
```

Comment:
debating if I should make all these functions take in `square` too.

Reply:
Wait a moment, why does this function take a `&self` parameter?

Reply:
That's definitely a function you wrote. Maybe a quick fix?